### DISCUSSION TITLE
Why am I getting "no images defined"?

### DISCUSSION THREAD

**Discussion by pfuntner** [ImageMagick/ImageMagick#6587] [Help] [answered]
### ImageMagick version

7.1.1-16

### Operating system, version and so on

Ubuntu 22

### Description

I built and installed from source code on Ubuntu 22 and I'm getting _no images defined_ when I try to run the `convert` command.

```
$ convert --version
Version: ImageMagick 7.1.1-16 (Beta) Q16-HDRI x86_64 0ca2131d1:20230824 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): 
Compiler: gcc (11.4)
$ convert logo: logo.gif
$ convert -rotate 90 logo.gif
convert: no images defined `logo.gif' @ error/convert.c/ConvertImageCommand/3354.
$
```

The `-debug` command did not reveal any more information.

Am I doing something wrong?

**Comment by fmw42** [comment, marked as answer]
> convert -rotate 90 logo.gif

That command has no input image

`convert logo: -rotate 90 logo.gif`

would be a correct command.

Correct syntax is

`convert input_image settings commands output_image`

see https://imagemagick.org/Usage/basics/#cmdline

**Reply by pfuntner** [reply]
Thanks
